compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -1031,12 +1031,9 @@ fn assemble_candidates_from_impls<'cx, 'tcx>(
                     {
                         candidate_set.mark_ambiguous();
                         true
-                    } else if obligation.predicate.args.type_at(0).to_opt_closure_kind().is_some()
-                        && obligation.predicate.args.type_at(1).to_opt_closure_kind().is_some()
-                    {
-                        true
                     } else {
-                        false
+                        obligation.predicate.args.type_at(0).to_opt_closure_kind().is_some()
```

Comment:
not sure whether this is really better :shrug:

Reply:
I think `if cond { true } else { false }` is kinda silly :sweat_smile:

Reply:
alone yes, but in an `else if` chain, keeping it consistent can be nicer